tests/src/test/java/alluxio/shell/command/FreeCommandTest.java

```diff
@@ -56,10 +69,25 @@ public void freeWildCardTest() throws IOException, AlluxioException {
     Assert.assertTrue(isInMemoryTest("/testWildCards/foobar4"));
 
     ret = mFsShell.run("free", "/testWild*/*/");
-    CommonUtils.sleepMs(null,
-        configuration.getInt(Constants.WORKER_BLOCK_HEARTBEAT_INTERVAL_MS) * 2 + 10);
+    triggerWorkerHeartbeats();
     Assert.assertEquals(0, ret);
     Assert.assertFalse(isInMemoryTest("/testWildCards/bar/foobar3"));
     Assert.assertFalse(isInMemoryTest("/testWildCards/foobar4"));
   }
+
+  private void triggerWorkerHeartbeats() {
+    // Execute the blocks free, which needs two heartbeats. Make sure there is some time delay
+    // between two heartbeats to make sure worker got time to generate block removal reports.
+    HeartbeatScheduler.schedule(HeartbeatContext.WORKER_BLOCK_SYNC);
+    try {
+      Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
+          TimeUnit.SECONDS));
+      CommonUtils.sleepMs(50);
```

Comment:
is there a way to get rid of this sleep? This could be an opportunity for non-deterministic tests?